src/utils/structure.js

```diff
@@ -1,50 +1,128 @@
-const items = [
-  {
-    description: 'Doctor appointment at 10AM',
-    completed: false,
-    index: 0,
-  },
-  {
-    description: 'Visit friends home at 12PM',
-    completed: false,
-    index: 1,
-  },
-  {
-    description: 'Complete project for Microverse',
-    completed: false,
-    index: 2,
-  },
-  {
-    description: 'Take programming tutorials on udemy',
-    completed: false,
-    index: 3,
-  },
-  {
-    description: 'Dinner plan with family at hotel',
-    completed: false,
-    index: 4,
-  },
-];
-
-function displayTask() {
+/* eslint-disable no-use-before-define */
+import userInteraction from './interaction.js';
+
+const items = [];
+const form = document.querySelector('.form-element');
+const displayTask = () => {
   const container = document.querySelector('.list-container');
+  form.addEventListener('submit', (event) => {
+    event.preventDefault();
+    const data = form.elements[0].value;
+    const object = {
+      description: data,
+      completed: false,
+      index: items.length + 1,
+    };
+    form.reset();
+    items.push(object);
+    localStorage.setItem('itemsLocal', JSON.stringify(items));
+    render();
+  });
+  const clearElement = document.createElement('p');
+  clearElement.className = 'clear';
+  clearElement.innerText = 'Clear all completed';
+  container.appendChild(clearElement);
+  const clear = document.querySelector('.clear');
   const list = document.createElement('ul');
+  list.className = 'ul-list';
+  container.insertBefore(list, clear);
+};
+
+const removeItem = () => {
+  const button = document.querySelectorAll('.fa-trash-alt');
+  button.forEach((item) => {
+    const parent = item.parentNode;
+    const superParent = parent.parentNode;
+    const index = Array.prototype.indexOf.call(superParent.children, parent);
+    const listInput = parent.firstChild;
+    item.addEventListener('click', () => {
+      if (listInput.hasAttribute('checked')) {
+        parent.remove();
+        items.splice(index, 1);
+      }
+      for (let i = 0; i < items.length; i += 1) {
+        items[i].index = i + 1;
+      }
+      localStorage.setItem('itemsLocal', JSON.stringify(items));
+    });
+  });
+};
+
+const clearList = () => {
+  const getClearElement = document.querySelector('.clear');
+  getClearElement.addEventListener('click', () => {
+    for (let i = 0; i < items.length; i += 1) {
+      if (items[i].completed) {
+        items.splice(i, 1);
+      }
+    }
+    for (let i = 0; i < items.length; i += 1) {
+      items[i].index = i + 1;
+    }
+    localStorage.setItem('itemsLocal', JSON.stringify(items));
+    render();
+  });
+};
```

Comment:
- [ ] It seems like the "Clear all completed" functionality it's not working properly. When I click on the button the tasks that are marked as completed remain in the list.

> A function for the "Clear all completed" button using the `Array.filter()` method is implemented.

![todo-list](https://user-images.githubusercontent.com/79114233/144674014-8d27d246-59ba-4fa3-b690-f699d275edb0.gif)